pkg/controller/orchestrator/orchestrator_reconcile_test.go

```diff
@@ -0,0 +1,354 @@
+
+				inst = orcClient.AddInstance(cluster.GetClusterAlias(), cluster.GetPodHostname(3),
+					false, -1, false, true)
+				inst.Key.Port = 3306
+				inst.MasterKey = orc.InstanceKey{Hostname: cluster.GetPodHostname(2), Port: 3306}
+
+				inst = orcClient.AddInstance(cluster.GetClusterAlias(), cluster.GetPodHostname(4),
+					false, -1, false, true)
+				inst.Key.Port = 3306
+				inst.MasterKey = orc.InstanceKey{Hostname: cluster.GetPodHostname(3), Port: 3306}
+
+				// Topology:
+				// 0 < - 1
+				//    \ - 2 < - 3 < - 4
+
+				var insts InstancesSet
+				insts, _ = orcClient.Cluster(cluster.GetClusterAlias())
+
+				// should determin the master as 0
+				master := insts.DetermineMaster()
+				Expect(master.Key.Hostname).To(Equal(cluster.GetPodHostname(0)))
+			})
+
+			It("existence of multiple masters", func() {
+				//AddInstance signature: cluster, host string, master bool, lag int64, slaveRunning, upToDate bool
+				inst := orcClient.AddInstance(cluster.GetClusterAlias(), cluster.GetPodHostname(0),
+					false, -1, false, true)
+				inst.Key.Port = 3306
+				inst.MasterKey = orc.InstanceKey{Hostname: cluster.GetPodHostname(5), Port: 3306}
+				inst.IsCoMaster = true
+
+				inst = orcClient.AddInstance(cluster.GetClusterAlias(), cluster.GetPodHostname(1),
+					false, -1, false, true)
+				inst.Key.Port = 3306
+				inst.MasterKey = orc.InstanceKey{Hostname: cluster.GetPodHostname(0), Port: 3306}
+
+				inst = orcClient.AddInstance(cluster.GetClusterAlias(), cluster.GetPodHostname(2),
+					false, -1, false, true)
+				inst.Key.Port = 3306
+				inst.MasterKey = orc.InstanceKey{Hostname: cluster.GetPodHostname(0), Port: 3306}
+
+				inst = orcClient.AddInstance(cluster.GetClusterAlias(), cluster.GetPodHostname(3),
+					false, -1, false, true)
+				inst.Key.Port = 3306
+				inst.MasterKey = orc.InstanceKey{Hostname: cluster.GetPodHostname(2), Port: 3306}
+
+				inst = orcClient.AddInstance(cluster.GetClusterAlias(), cluster.GetPodHostname(4),
+					false, -1, false, true)
+				inst.Key.Port = 3306
+				inst.MasterKey = orc.InstanceKey{Hostname: cluster.GetPodHostname(5), Port: 3306}
+
+				inst = orcClient.AddInstance(cluster.GetClusterAlias(), cluster.GetPodHostname(5),
+					false, -1, false, true)
+				inst.Key.Port = 3306
+				inst.MasterKey = orc.InstanceKey{Hostname: cluster.GetPodHostname(0), Port: 3306}
+				inst.IsCoMaster = true
+
+				// Topology:
+				//  0 <- 1
+				//  |  \_ 2 <- 3
+				//  5 <- 4
+
+				var insts InstancesSet
+				insts, _ = orcClient.Cluster(cluster.GetClusterAlias())
+
+				// should not determin any master because there are two masters
+				master := insts.DetermineMaster()
+				Expect(master).To(BeNil())
+
+			})
+
+			It("no instances", func() {
+				var insts InstancesSet
+				insts, _ = orcClient.Cluster(cluster.GetClusterAlias())
+
+				master := insts.DetermineMaster()
+				Expect(master).To(BeNil())
+			})
+
+			Describe("orc updater unit tests", func() {
```

Comment:
This should be a new top level suite.

Reply:
but I'm using `cluster` from previous `Describe`.